src/transformers/generation/logits_process.py

```diff
@@ -520,6 +520,82 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
         return scores_processed
 
 
+class MinPLogitsWarper(LogitsWarper):
+    """
+    [`LogitsWarper`] that performs min-p, i.e. keeps all tokens that are above a minimum probability, scaled by the
+    probability of the most likely token. As a result, the filter becomes more agressive in the presence of
+    high-probability tokens, which is a sign of a confident output that we shouldn't deviate from.
+
+    Often used together with [`TemperatureLogitsWarper`]. Used as an alternative to [`TopPLogitsWarper`] and
+    [`TopKLogitsWarper`].
+
+    Created by @menhguin and @kalomaze (github handles). Code adapted from [this external PR](https://github.com/oobabooga/text-generation-webui/pull/4449/files)
+
+    Args:
+        min_p (`float`):
+            Minimum token probability, which will be scaled by the probability of the most likely token. It must be a
+            value between 0 and 1, with typical values in the 0.01-0.2 range.
```

Comment:
I personally recommend:

```suggestion
            Minimum token probability, which will be scaled by the probability of the most likely token. It must be a
            value between 0 and 1, Typical values are in the 0.01-0.2 range, comparable to using top p=0.99 - 0.8                         
            (so use the opposite of normal top p values).
```
IDK how else to phrase that, a lot of people just get tripped up using Min P = 0.9 thinking it's like Top P = 0.9 and then they get disapppointed that it's too deterministic.